.github/workflows/docker_hub.yml

```diff
@@ -71,11 +74,12 @@ jobs:
         if: env.JOB_DONE == 'false'
 
       - name: Login to Docker Hub
+        id: login
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-        if: env.JOB_DONE == 'false'
+        if: env.JOB_DONE == 'false' && env.CAN_DEPLOY == 'true'
 
```

Comment:
`env.CAN_LOGIN`?

Reply:
> `env.CAN_LOGIN`?

I've run a [test](https://github.com/soehms/sage/actions/runs/9195102984/job/25290187494#step:7:8) on this branch merged with the branch of #38049 after I changed `CAN_DEPLOY` -> `CAN_LOGIN`. This has been successful producing the [beta6](https://hub.docker.com/layers/sagemath/sagemath/10.4.beta6/images/sha256-27ca25158100d1a72ba3886946abcee4ee037b7150628383667633f16c58d195?context=repo) image on Docker Hub.

Reply:
Thanks @kwankyu @soehms, I've fixed this in ffabaaf6514

Reply:
The PR is still a draft!